packages/ember-views/tests/views/view/init_test.js

```diff
@@ -30,7 +30,7 @@ test("registers view in the global views hash using layerId for event targeted",
 
 QUnit.module("EmberView.createWithMixins");
 
-test("should warn if a non-array is used for classNames", function() {
+test("should warn if a computed property is used for classNames", function() {
```

Comment:
I should have said: I can do a separate PR to test that `Ember.assert` catches cases in which `classNames` is neither an array nor a computed property (e.g., a string, integer, regex, etc.).

Off the top of my head, I'd guess that the only way to do this is to call `Ember.assert()` prior to calling `this._super()` in the view's `init()`. That way, Ember won't have had a chance to concatenate properties yet, and so `classNames` won't have been converted to an array yet.